cpp/src/copying/copy.cu

```diff
@@ -17,39 +17,187 @@
+   column_device_view bool_mask_device = *bool_mask_device_p;                                    
+   
    auto filter = [bool_mask_device] __device__ (cudf::size_type i) { return bool_mask_device.element<cudf::experimental::bool8>(i); };
+   return cudf::experimental::type_dispatcher(lhs.type(),                                             
+                                             copy_if_else_functor{},
+                                             lhs, rhs, boolean_mask.size(),
+                                             left_nullable, right_nullable,
+                                             filter,
+                                             mr, stream);
+}
+
+}; // namespace anonymous
+
+std::unique_ptr<column> copy_if_else( column_view const& lhs, column_view const& rhs, column_view const& boolean_mask,
+                                      rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+{
+   CUDF_EXPECTS(boolean_mask.size() == lhs.size(), "Boolean mask column must be the same size as lhs and rhs columns");
+   CUDF_EXPECTS(lhs.size() == rhs.size(), "Both columns must be of the size"); 
+   return copy_if_else(*column_device_view::create(lhs), *column_device_view::create(rhs), lhs.has_nulls(), rhs.has_nulls(), boolean_mask, mr, stream);
+}
+
+std::unique_ptr<column> copy_if_else( scalar const& lhs, column_view const& rhs, column_view const& boolean_mask,
+                                      rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+{
+   CUDF_EXPECTS(boolean_mask.size() == rhs.size(), "Boolean mask column must be the same size as rhs column");
+   return copy_if_else(lhs, *column_device_view::create(rhs), !lhs.is_valid(), rhs.has_nulls(), boolean_mask, mr, stream);
+}
+
+std::unique_ptr<column> copy_if_else( column_view const& lhs, scalar const& rhs, column_view const& boolean_mask,
+                                      rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+{
+   CUDF_EXPECTS(boolean_mask.size() == lhs.size(), "Boolean mask column must be the same size as lhs column");
+   return copy_if_else(*column_device_view::create(lhs), rhs, lhs.has_nulls(), !rhs.is_valid(), boolean_mask, mr, stream);
+}
 
-   return copy_if_else(lhs, rhs, filter, mr, stream);
+std::unique_ptr<column> copy_if_else( scalar const& lhs, scalar const& rhs, column_view const& boolean_mask,
+                                      rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+{
+   return copy_if_else(lhs, rhs, !lhs.is_valid(), !rhs.is_valid(), boolean_mask, mr, stream);
 }
 
-}  // namespace detail
+}; // namespace detail
 
 std::unique_ptr<column> copy_if_else( column_view const& lhs, column_view const& rhs, column_view const& boolean_mask,
-                                      rmm::mr::device_memory_resource *mr)
+                                      rmm::mr::device_memory_resource* mr)
+{      
+   return detail::copy_if_else(lhs, rhs, boolean_mask, mr, (cudaStream_t)0);   
```

Comment:
Is passing`(cudaStream_t)0` necessary? `0` is default argument defined in header file.